### DISCUSSION TITLE
[Enhancement] Song picker

### DISCUSSION THREAD

**Discussion by sircharlo** [sircharlo/meeting-media-manager#845] [Ideas] [answered]
While going through scenarios, we came across these two. I'd like your thoughts @mtdvlpr 

1. The speaker arrives right before the meeting starts. He hasn't had the time to confirm his song beforehand. He lets the AV operator know his song number 1 minute before the chairman announces it from the platform.
2. The speaker has given the wrong song number to the AV operator and chairman. He later realises his mistake, and lets the chairman know. The chairman forgot to let the AV operator know, and the AV operator finds out the correct song number as the chairman is announcing it. 

For these cases, would it be practical to have an item in the menu on the top left, that would allow the AV operator to:
- pick a song from the media language (main, or fallback if main is not available)
- add a toggle to have the option to stream the song rather than downloading it
- if "stream" is enabled, add a toggle to have the option to automatically click on "play" after adding the media item to the list (to allow to start playback immediately, in "urgent" situations)
- add that song automatically as the first media item in the media list

For scenario 1, the song could be downloaded (in other words, basically following the same workflow as if the AV operator had clicked on the meeting day's tile on the main page) but would skip the congregation server (if configured), as it's needed ASAP. 

For scenario 2, the AV operator would select the "stream" option, and would then also activate the "auto-click on play" option. The song would then start playing immediately, from the remote URL. (Not ideal, but better than having the congregation wait 10-30 seconds for the file to finish downloading before being able to play it.)

**Comment by mtdvlpr** [comment, marked as answer]
I can definitely see these scenarios happen, so I like these suggestions! Scenario 1 sounds straight-forward enough, a song selector in the topbar of the presentation mode. I don't yet see the full picture of scenario 2. You mention stream and auto-click on play options. Where do you see these options?

What I now envision is a song picker, which would add the song immediately to the top of the media list similarly to the other media, but with a download and stream button. Download button would become a loading indicator and then disappear. The stream button would stream the song, or if it has been downloaded, just play the song. But I'm curious what you had in mind

Btw, I'm now wondering. We haven't implemented the fallback language into the song picker at the manage media screen. Irish for example has about 20 songs or so, so then the fallback language could fill the rest of the list (fixed in #846).

**Reply by sircharlo** [reply]
> What I now envision is a song picker, which would add the song immediately to the top of the media list similarly to the other media, but with a download and stream button. Download button would become a loading indicator and then disappear. The stream button would stream the song, or if it has been downloaded, just play the song. But I'm curious what you had in mind

Oh I like that! Simpler than what I had envisioned. Song picker, gets added to media list, two buttons: download and stream/play.

**Comment by mtdvlpr** [comment]
I would say this is something for v23.1.0 and 22.12.1 is ready to be released now.

**Reply by sircharlo** [reply]
Sounds good, I'll release v22.12.1 then.

**Comment by mtdvlpr** [comment]
I feel like we're reaching the end of features/improvements to implement...discussions and issues have been going down significantly. I think we are allowed to be a little proud of ourselves for getting to this level of stability :)

**Reply by sircharlo** [reply]
Yeah I feel that too. It feels like now there will mostly only be bugfixes and tweaks here and there, but all in all the app is solid and a joy to use!

**Comment by mtdvlpr** [comment]
@sircharlo, see #857 for an initial setup :)
